File: backend/eventticketingplatform/src/main/java/com/nipapager/eventticketingplatform/review/service/ReviewService.java
package com.nipapager.eventticketingplatform.review.service;

import com.nipapager.eventticketingplatform.response.Response;
import com.nipapager.eventticketingplatform.review.dto.ReviewDTO;

import java.util.List;

/**
 * Service interface for review operations
 */
public interface ReviewService {

    Response<ReviewDTO> createReview(ReviewDTO reviewDTO);

    Response<List<ReviewDTO>> getReviewsByEventId(Long eventId);

    Response<ReviewDTO> getReviewById(Long id);

    Response<ReviewDTO> updateReview(Long id, ReviewDTO reviewDTO);

    Response<Void> deleteReview(Long id);

    // Get reviews by current user
    Response<List<ReviewDTO>> getMyReviews();
}

File: frontend/event-ticketing-app/src/api/reviewService.ts
import api from './axios';

export interface ReviewDTO {
  id: number;
  userId: number;
  userName: string;
  eventId: number;
  eventName: string;
  rating: number;
  comment?: string;
  createdAt: string;
  updatedAt: string;
}

export interface ReviewSummaryDTO {
  averageRating: number;
  totalReviews: number;
}

export interface CreateReviewRequest {
  eventId: number;
  rating: number;
  comment?: string;
}

const reviewService = {
  // Create review
  createReview: async (request: CreateReviewRequest): Promise<ReviewDTO> => {
    const response = await api.post('/reviews', request);
    return response.data.data;
  },

  // Update review
  updateReview: async (reviewId: number, request: CreateReviewRequest): Promise<ReviewDTO> => {
    const response = await api.put(`/reviews/${reviewId}`, request);
    return response.data.data;
  },

  // Get reviews for an event
  getEventReviews: async (eventId: number): Promise<ReviewDTO[]> => {
    const response = await api.get(`/reviews/event/${eventId}`);
    return response.data.data;
  },

  // Get review summary for an event
  getReviewSummary: async (eventId: number): Promise<ReviewSummaryDTO> => {
    const response = await api.get(`/reviews/event/${eventId}/summary`);
    return response.data.data;
  },

  // Get current user's review for an event
  getUserReviewForEvent: async (eventId: number): Promise<ReviewDTO | null> => {
    const response = await api.get(`/reviews/event/${eventId}/my-review`);
    return response.data.data;
  },

  // Get all my reviews
  getMyReviews: async (): Promise<ReviewDTO[]> => {
    const response = await api.get('/reviews/my-reviews');
    return response.data.data;
  },

  // Delete review
  deleteReview: async (reviewId: number): Promise<void> => {
    await api.delete(`/reviews/${reviewId}`);
  },
};

export default reviewService;